src/index.js

```diff
@@ -155,28 +156,32 @@ const compile = (schema, root, reporter, opts, scope, basePathRoot) => {
   fun.write('function validate(data) {')
   // Since undefined is not a valid JSON value, we coerce to null and other checks will catch this
   fun.write('if (data === undefined) data = null')
-  if (reporter === true) fun.write('validate.errors = null')
+  if (optIncludeErrors) fun.write('validate.errors = null')
   fun.write('let errors = 0')
 
   const basePathStack = basePathRoot ? [basePathRoot] : []
-  const visit = (allErrors, reporter, name, node, schemaPath) => {
-    const rule = (...args) => visit(allErrors, reporter, ...args)
+  const visit = (allErrors, includeErrors, name, node, schemaPath) => {
+    const rule = (...args) => visit(allErrors, includeErrors, ...args)
     const subrule = (...args) => visit(true, false, ...args)
+    const writeError = (format, ...params) => {
```

Comment:
Somewhat confusing having both this function and `error` function in the same scope. Maybe we should either rename them both, or move this one inside `error`'s local scope

Reply:
Can't come up with anything good :-(
Renamed to `writeErrorObject`, which should it at least less similar to `error`.